feat: add CU worker binary

monoid opened the pull request.

Commit:
Move `cu` to a separate crate

The crate doesn't compile yet, but `ccp` dependencies are made explicit.
`ccp` crate is not corrected too.

Commit:
ccp-cu crate compiles

Commit:
Fix `ccp` compilation

Commit:
Stale test fix

Commit:
Remove unused deps in `ccp` and `ccp_cu`

With `cargo machete`.

Commit:
Re-export `anyhow::Error` as `AnyhowError`

Commit:
Make fmt happy

Commit:
ccp-cu-main sketch

Commit:
Make `cu_id` optional in many places

Commit:
make fmt happy

Commit:
JSON-RPC transport for proofs

Commit:
Use single-thread runtime in ccp-cu-main

Commit:
Add cu_main/README.md

Commit:
Implement tonic client instead of jsonrpsee

The latter doesn't seem to be able to work with Unix domain sockets.

Commit:
Fix to use new client in CU worker

Commit:
Refactor CU worker

Commit:
Basic dockerfile

Commit:
Update cu_main README.md

Commit:
Merge branch 'main' into feat/BRND-63-ccp-cu-binary

monoid commented on cu_main/docker/Dockerfile:
```diff
@@ -0,0 +1,46 @@
+COPY cu_main/docker/Config.default.toml /Config.toml
+COPY cu_main/docker/entrypoint.sh /entrypoint.sh
+USER ccp_user
+ENTRYPOINT ["/usr/local/bin/ccp_cu"]
```
It should be `/entrypoint.sh`.

monoid commented on Cargo.toml:
```diff
@@ -18,6 +18,7 @@ members = [
     "crates/test-utils",
     "crates/utils",
     "main",
+    "cu_main", "crates/cu-rpc-client",
```
Fix formatting.

monoid commented on crates/config/src/cu_config.rs:
```diff
@@ -0,0 +1,65 @@
+#[derive(Clone, Debug, PartialEq)]
+pub struct CUWorkerConfig {
+    pub optimizations: Optimizations,
+    // TODO remove
```
Remove.

Commit:
Make clippy and fmt happy

Commit:
Add simpler log config for CU worker

Commit:
Fix Cargo.toml formatting

Commit:
Rename `cu_main` to `cu_worker` and binary

from `ccp_cu` to `ccp_cu_worker`.

Commit:
Worker may connect to both HTTP(S) and UDS

Commit:
gRPC API accepts multiple proofs

Commit:
Merge branch 'main' into feat/BRND-63-ccp-cu-binary

gurinderu commented on crates/cu-rpc-client/proto/collector.proto:
```diff
@@ -0,0 +1,27 @@
+package cu_rpc_client;
+
+message EpochParameters {
+    string difficulty = 1;
```
Lets use bytes everywhere for the byte arrays.

monoid replied:
I think protobuf suxx in Rust projects and should be removed. It gives you foreign, unergonomic, un-Rusty types.

monoid replied:
Using bytes would involve writing more code without real benefits.

gurinderu commented on cu_worker/src/main.rs:
```diff
@@ -0,0 +1,122 @@
+    global_nonce: GlobalNonce,
+    difficulty: Difficulty,
+    cu_id: CUID,
+    upstream_url: String,
```
Url ?

monoid replied:
Renamed to upstream.

Commit:
Merge branch 'main' into feat/BRND-63-ccp-cu-binary

Commit:
Document new varialbe `TEMP_DIR`

Commit:
Rename `UPSTREAM_URL` env to `UPSTREAM`

Commit:
Cargo fixes

Commit:
Improve RPC client

0. Rename RPC's generated type `Proof` to `RpcProof` to make it more
   distinct from CCP `Proof`/`RawProof`.
1. Add `From` for converting `RawProof` to `RpcProof` and `TryFrom` for
   converting `RpcProof` to `RawProof`.
3. Reexport certain useful types.

Commit:
Fix tests

Commit:
Rename RPC's `EpochParameters` to `RpcEpochParameters`

Commit:
More reexports

gurinderu commented on crates/cu-rpc-client/src/lib.rs:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    pub async fn auto(endpoint: String) -> Result<Self, tonic::transport::Error> {
+        if endpoint.starts_with("http://") || endpoint.starts_with("https://") {
```
grpc url typically start with `grpc://`

Commit:
Remove unused entrypoint.sh

Commit:
Add grpc to the recognized protocol list

Commit:
Implement client retry

Commit:
Add tracing_log to output ccp events

Commit:
Add missing file

enjenjenje committed:
chore: CU Worker cicd (#179)

* cu worker cicd

* cu worker cicd

* fix

* try recursive checkout

* cu-worker process redone

* cu-worker process redone

* cu-worker process redone

* fixed lint

* added new line

* added new line

Commit:
Merge branch 'main' into feat/BRND-63-ccp-cu-binary

Commit:
Move network code for a cu_worker to separate crate

To make it testable.

Commit:
cu-worker RetryClient test

gurinderu commented on .github/workflows/artifact.yml:
````diff
@@ -203,4 +203,4 @@ jobs:
           docker login docker.fluence.dev
           ```
-          SNAPSHOT
+          SNAPSHOT
````
\n at eof

Commit:
`RetryClient` quits if sender is dead

Previously, `eyre::Error` was returned.

Commit:
Add final newline to artifact.yaml
